morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -1144,12 +1151,58 @@ protected void ensureIndex(final MappedClass mc, final DBCollection dbColl, fina
         }
 
         LOG.debug(format("Creating index for %s with keys:%s and opts:%s", dbColl.getName(), keys, opts));
-        dbColl.createIndex(keys, opts);
+        createMongoIndex(dbColl, keys, opts);
     }
 
-    protected void ensureIndex(final DBCollection dbColl, final DBObject keys, final DBObject options) {
+    protected void ensureIndex(final DBCollection dbColl, final DBObject keys, final BasicDBObject options) {
         LOG.debug(format("Creating index for %s with keys:%s and opts:%s", dbColl.getName(), keys, options));
-        dbColl.createIndex(keys, options);
+        createMongoIndex(dbColl, keys, options);
+    }
+
+    private void createMongoIndex(final DBCollection dbColl, final DBObject keys, final DBObject options) {
+        try {
+            dbColl.createIndex(keys, options);
+        } catch (MongoCommandException ex) {
+            if (options.containsField("dropOutdated") && getBoolean(options, "dropOutdated", false)) {
```

Comment:
A cast to BasicDBObject is sufficient here and then you'd have access to these methods defined there.

Also, checking contains first is redundant in this case since the default on the get is false.